content/blog/2016-09-14-interview-with-digitalocean.md

```diff
@@ -0,0 +1,92 @@
+organizational process (which was hard to change/enforce). Other teams were
+frustrated with the query language and the visualization tools available at the
+time. I was chatting with Julius Volz about push vs poll metric systems and was
+sold in wanting to try Prometheus when I saw that I would really be in control
+of my SLA when I get to determine what I’m polling and how frequently. Plus, I
+really really liked the query language.
+
+## How did you transition?
+
+We were gathering metrics via Collectd sending to OpenTSDB. Installing the Node
+Exporter in parallel with our already running Collectd setup allowed us to
+start experimenting with Prometheus. We also created a custom exporter to
+expose Droplet metrics. Soon, we had feature parity with our OpenTSDB service
+and started turning off Collectd and then turned off the OpenTSDB cluster.
+
+People really liked Prometheus and the visualization tools that came with it.
+Suddenly, my small metrics team had a backlog that we couldn’t get to fast
+enough to make people happy, and instead of providing and maintaining
+Prometheus for people’s services, we looked at creating tooling to make it as
+easy as possible for other teams to run their own Prometheus servers and to
+also run the common exporters we use at the company.
+
+Some teams have started using Alertmanager, but we still have a concept of
+polling Prometheus from our existing monitoring tools.
+
+## What improvements have you seen since switching?
+
+We’ve improved our insights on hypervisor machines. The data we could get out
+of Collectd and Node Exporter is about the same, but it’s much easier for our
+team of golang developers to create a new custom exporter that exposes data
+specific to the services we run on each hypervisor.
+
+We’re exposing better application metrics. It’s easier to learn and teach how
+to create a Prometheus metric that can be aggregated correctly later. With
+Graphite it’s easy to create a metric that can’t be aggregated in a certain way
+later because the dot-separated-name wasn’t structured right.
+
+Creating alerts is much quicker and simpler than what we had before, plus in a
+language that is familiar. This has empowered teams to create better alerting
+for the services they know and understand because they can iterate quickly.
+
+## What do you think the future holds for DigitalOcean and Prometheus?
+
+We’re continuing to look at how to make collecting metrics as easy as possible
+for teams at DigitalOcean. Right now teams are running their own Prometheus
+servers for the things they care about, which allowed us to gain observability
+we otherwise wouldn’t have had as quickly. But, not every team should know how
+to run Prometheus. We’re looking at what we can do to make Prometheus as
+automatic as possible so that teams can just concentrate on what queries and
+alerts they want on their services and databases.
+
+We’re also looking at what we can do to have long-term data storage while
```

Comment:
This is alluding to Vulcan which wasn't open sourced at the time of writing. We can name it and link to it now.

```markdown
We also created [Vulcan](https://github.com/digitalocean/vulcan) so that we have long-term data storage while
retaining the Prometheus Query Language that we have built tooling around and
trained people how to use.
```